packages/xo-server/src/xapi/index.js

```diff
@@ -1924,7 +1924,12 @@ export default class Xapi extends XapiBase {
   async _deleteVdi(vdiRef) {
     log.debug(`Deleting VDI ${vdiRef}`)
 
-    await this.call('VDI.destroy', vdiRef)
+    await this.call('VDI.destroy', vdiRef).catch(error => {
+      if (error.code !== 'HANDLE_INVALID') {
+        log.error('_deleteVdi', { error })
```

Comment:
Why adding a log here?

Reply:
No particular reason, just a habit of logging error. I'll remove it.

Reply:
Leave it, it's not a bad idea, I was just wondering it came from :-)

Though you could add the ref of the VDI in the data bin.